### PR TITLE
refactor: provide cyclone base images to build images

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #https://github.com/caicloud/cyclone/issues/1390

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @hyy0322 @chechaoa 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/cicd/cd/Dockerfile.arm64 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM arm64v8/alpine:3.8
+FROM caicloud/cyclone-base-alpine:v1.0.0-arm64v8
```
Comment (hyy0322) [inline build/cicd/cd/Dockerfile.arm64:1]: arm64v8 是不是不要放在 tag 里区分，我感觉放在 project 里区分比较好一些

Comment (hyy0322) [reply build/cicd/cd/Dockerfile.arm64:1]: 打扰了 dockerhub 不支持

**Comment by hyy0322** [comment]
LGTM

File: Makefile [hunk 2]
```diff
@@ -37,11 +38,17 @@ IMAGE_SUFFIX ?= $(strip )
 
 # REGISTRY ?= docker.io/library
```
Comment (supereagle) [inline Makefile:39]: Remove this?

File: Makefile [hunk 3]
```diff
@@ -37,11 +38,17 @@ IMAGE_SUFFIX ?= $(strip )
...
 # Container registry for target images.
-REGISTRY ?= docker.io/library
+REGISTRY ?= docker.io/caicloud
 
-# Container registry for base images.
+# Container registry for build base images.
 BASE_REGISTRY ?= docker.io/library
```
Comment (supereagle) [inline Makefile:44]: Why not update this?

File: Makefile [hunk 4]
```diff
@@ -37,11 +38,17 @@ IMAGE_SUFFIX ?= $(strip )
...
 
+# Container registry for cyclone target base images.
+CYCLONE_BASE_REGISTRY ?= docker.io/caicloud
```
Comment (supereagle) [inline Makefile:47]: I think not need this, can reuse the `REGISTRY`, as target images and base images will been in the same registry
.

Comment (supereagle) [inline Makefile:47]: I think not need this, can reuse the `REGISTRY`, as the target images and base images will been in the same registry.

File: Makefile [hunk 5]
```diff
@@ -37,11 +38,17 @@ IMAGE_SUFFIX ?= $(strip )
...
+
+# Version of the cyclone base images.
+CYCLONE_BASE_VERSION ?= v1.0.0
```
Comment (supereagle) [inline Makefile:50]: The prefix `CYCLONE_` is not necessary.

```suggestion
BASE_VERSION ?= v1.0.0
```

File: Makefile (hunk 2 above)
Comment (zhujian7) [reply Makefile:39]: ok

File: Makefile (hunk 4 above)
Comment (zhujian7) [reply Makefile:47]: `CYCLONE_BASE_REGISTRY` does not need updating frequently, it's used as base images in the Dockerfiles, and these Dockerfiles Base images will not be updated frequently.

File: Makefile (hunk 5 above)
Comment (zhujian7) [reply Makefile:49]: In order to keep consistent with `CYCLONE_BASE_REGISTRY`.

**Pull request merged**